ge/org.osate.ge.ba/src/org/osate/ge/ba/ui/properties/BehaviorTransitionPropertySection.java

```diff
@@ -95,56 +101,199 @@ public void createControls(final Composite parent, final TabbedPropertySheetPage
+				setControlsToMultipleSelected();
+			} else {
+				final BehaviorTransition behaviorTransition = (BehaviorTransition) selectedBoc.getBusinessObject();
+				final TransactionalEditingDomain editingDomain = (TransactionalEditingDomain) AdapterFactoryEditingDomain
+						.getEditingDomainFor(behaviorTransition);
+				final IProject project = ProjectUtil.getProjectForBoOrThrow(behaviorTransition);
+				final XtextResource xtextResource = getXtextResource(behaviorTransition)
+						.orElseThrow(() -> new RuntimeException("resource must be XtextResource"));
+				final IXtextDocument xtextDocument = getXtextDocument(behaviorTransition).orElse(null);
+				final String sourceText = BehaviorAnnexXtextUtil.getText(xtextDocument, xtextResource);
+				// Controls for editing dispatch conditions
+				createConditionEditingControls(behaviorTransition, sourceText, project, editingDomain, xtextDocument,
+						xtextResource);
+				// Controls for editing action blocks
+				createActionBlockEditingControls(behaviorTransition, sourceText, project, editingDomain, xtextDocument,
+						xtextResource);
+			}
+		} else {
+			disposeEditControls();
+		}
 
 		// Layout controls for refresh
+		layout();
+	}
+
+	private static void createEditingControls(final EmbeddedTextEditor controls,
+			final SelectionAdapter selectionAdpater,
+			final IProject project, final EmbeddedTextValue textValue) {
+		controls.addSelectionListener(selectionAdpater);
+		controls.createXtextAdapter(project, textValue);
+	}
+
+	private void createConditionEditingControls(final BehaviorTransition behaviorTransition, final String sourceText,
+			final IProject project, final TransactionalEditingDomain editingDomain, final IXtextDocument xtextDocument,
+			final XtextResource xtextResource) {
+		final EmbeddedTextValue conditionTextValue = getConditionTextValue(behaviorTransition, sourceText);
+		final SelectionAdapter editConditionSelectionAdapter = getEditConditionSelectionAdapter(project,
+				conditionTextValue, behaviorTransition, editingDomain, xtextDocument, xtextResource);
+		createEditingControls(conditionEditingControls, editConditionSelectionAdapter, project, conditionTextValue);
+	}
+
+	private void createActionBlockEditingControls(final BehaviorTransition behaviorTransition, final String sourceText,
+			final IProject project, final TransactionalEditingDomain editingDomain, final IXtextDocument xtextDocument,
+			final XtextResource xtextResource) {
+		final EmbeddedTextValue actionTextValue = getActionBlockTextValue(behaviorTransition, sourceText);
+		final SelectionAdapter editActionBlockSelectionAdapter = getEditActionSelectionAdapter(project, actionTextValue,
+				behaviorTransition, editingDomain, xtextDocument, xtextResource, sourceText);
+		createEditingControls(actionBlockEditingControls, editActionBlockSelectionAdapter, project, actionTextValue);
+	}
+
+	private void refreshEditControls() {
+		// Dispose before refresh
+		disposeEditControls();
```

Comment:
Instead of having to dispose things manually before refreshing, the refresh() of the editing controls should do this.